packages/app/src/systems/SignUp/hooks/useSignUpStepper.ts

```diff
@@ -0,0 +1,32 @@
+import { useNavigate } from 'react-router-dom';
+
+import { STORAGE_KEY } from '../components/SignUpProvider';
+import { SignUpType } from '../machines/signUpMachine';
+
+import { Pages, Storage } from '~/systems/Core';
+
+export function useSignUpStepper() {
+  const navigate = useNavigate();
+  const isCreate = Storage.getItem(STORAGE_KEY) === SignUpType.create;
```

Comment:
we should use the state of the machine to control the steps, on the SwaySwap we had issues by controlling the steps using a different data from the machine state. As the storage got out for sync with the real state on the machine.

Reply:
In fact and just does this workaround to get things done, but it's necessary to make an entire refactor of this machine that's so bad, but I just put some kinda a "temporary" solution to don't waste so much time